ipn/ipnlocal/c2n.go

```diff
@@ -337,8 +337,15 @@ func handleC2NPostureIdentityGet(b *LocalBackend, w http.ResponseWriter, r *http
 			http.Error(w, err.Error(), http.StatusInternalServerError)
 			return
 		}
-
 		res.SerialNumbers = sns
+
+		if r.FormValue("hwaddrs") == "true" {
+			res.IfaceHardwareAddrs, err = posture.GetHardwareAddrs()
+			if err != nil {
+				http.Error(w, err.Error(), http.StatusInternalServerError)
```

Comment:
If we're going to hard error with these, let's specify where the error is coming from:

```suggestion
				http.Error(w, "hwaddrs: "+err.Error(), http.StatusInternalServerError)
```

But I think it might be nice to do something like returning a list of errors `res.Errors = append(res.Errors, "hwaddrs: "+err.Error())` so that we can retrieve serial numbers even if hwaddr fails? If we did this, we could omit the `?hwaddrs=true` query param.

Reply:
I am not a fan of adding errors to response messages unless we actually intend to parse them and do something differently. I agree that we probably should not hard fail on this, so I've changed this to simply log the error instead.

I'd still prefer this gated by a GET parameter just in case we find something unexpected (e.g. it crashing or slowing down the client). We can remove it as we get more confidence with this.